packages/core/src/components/Dropdown/Dropdown.jsx

```diff
@@ -95,7 +96,8 @@ export class Dropdown extends React.PureComponent {
           /* eslint-enable no-return-assign */
           {...selectProps}
         >
-          {optionElements}
+          {/* Render custom options if provided */
+          children || optionElements}
```

Comment:
This is one of those situations where we want either options or children, but theres no good way to have that be enforced with Proptypes. Because we almost always want options to be defined, I left options as a required prop, while children is optional.

Reply:
You could provide a function to the options proptype, but it gets kind of complicated:
```
options: function(props) {
  if (props.children) return new Error('Cannot provide options and children at the same time');

  if (!props.options) return new Error('options prop must be defined');

  if (!Array.isArray(props.options)) return new Error('options prop must b an array')

  if (!props.options.every(option => option.label && option.value)) return new Error('options array does not match required shape')
}
```

It would be easier if you could use the PropTypes validators within the custom validator function but it looks like you can't. So I think it's probably simplest as you have it

Reply:
Yeah I'll do this and just add some info in the react prop documentation!